sqlglot/helper.py

```diff
@@ -356,6 +356,15 @@ def flatten(values: t.Iterable[t.Iterable[t.Any] | t.Any]) -> t.Generator[t.Any,
             yield value
 
 
+def count_params(function: t.Callable) -> int:
```

Comment:
how expensive is this? if it's expensive can we do this at load time instead of run time?

Reply:
According to [the source](https://github.com/python/cpython/blob/ace008c531dd685a30c1dd68f9b5ba35f20171cf/Lib/inspect.py#L82-L90), seems like it's doing an instance check under the hood. Maybe it's ok?

Reply:
benchmark it,

import time

now  = time.time()
...